v0.8.4 dual stack static pod fails with runtime error

**Describe the bug**
kube-vip v0.8.4 dual stack static pod fails to run on any control plane.
After the IP addresses are activated a runtime error occurs.

**To Reproduce**
Generate a manifest, place it in /etc/kubernetes/manifests, change to super-admin.conf, initialize the cluster and change back to admin.conf, eg:
```bash
export INTERFACE=enp1s0
KVVERSION=v0.8.4
KUBEVIP="docker run --network host --rm ghcr.io/kube-vip/kube-vip:$KVVERSION"
export VIP=10.10.10.12,fd00:4269:6943::12

$KUBEVIP manifest pod \
    --interface $INTERFACE \
    --address $VIP \
    --controlplane \
    --arp \
    --leaseDuration 15 \
    --leaseRenewDuration 10 \
    --leaseRetry 2 \
    --leaderElection > kube-vip.yaml

scp kube-vip.yaml root@host:/etc/kubernetes/manifests
ssh root@host
sed -i 's#path: /etc/kubernetes/admin.conf#path: /etc/kubernetes/super-admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml
kubeadm init --control-plane-endpoint ...
sed -i 's#path: /etc/kubernetes/super-admin.conf#path: /etc/kubernetes/admin.conf#' /etc/kubernetes/manifests/kube-vip.yaml
```

**Expected behavior**
kube-vip v0.8.4 runs dual stack configuration without errors.

**Environment (please complete the following information):**
 - OS/Distro: Debian 12.7
 - Kubernetes Version: 1.31.1
 - Kube-vip Version: 0.8.4

**Screenshots**
Full log of kube-vip pod:
```
time="2024-10-23T06:27:57Z" level=info msg="Starting kube-vip.io [v0.8.4]"
time="2024-10-23T06:27:57Z" level=info msg="namespace [kube-system], Mode: [ARP], Features(s): Control Plane:[true], Services:[false]"
time="2024-10-23T06:27:57Z" level=info msg="prometheus HTTP server started"
time="2024-10-23T06:27:57Z" level=info msg="Using node name [cp1]"
time="2024-10-23T06:27:57Z" level=info msg="Starting Kube-vip Manager with the ARP engine"
time="2024-10-23T06:27:57Z" level=info msg="Beginning cluster membership, namespace [kube-system], lock name [plndr-cp-lock], id [cp1]"
I1021 06:27:57.016794       1 leaderelection.go:250] attempting to acquire leader lease kube-system/plndr-cp-lock...
I1021 06:27:57.030441       1 leaderelection.go:260] successfully acquired lease kube-system/plndr-cp-lock
time="2024-10-23T06:27:57Z" level=info msg="Node [cp1] is assuming leadership of the cluster"
time="2024-10-23T06:27:57Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [10.10.10.12/enp1s0]"
time="2024-10-23T06:27:57Z" level=info msg="Gratuitous Arp broadcast will repeat every 3 seconds for [fd00:4269:6943::12/enp1s0]"
time="2024-10-23T06:27:57Z" level=info msg="Broadcasting NDP update for fd00:4269:6943::12 (52:54:00:a3:42:ff) via enp1s0"
time="2024-10-23T06:27:57Z" level=info msg="ndp: &{false false true fd00:4269:6943::12 [0xc000446f80]}"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1783770]

goroutine 45 [running]:
github.com/kube-vip/kube-vip/pkg/vip.(*NdpResponder).SendGratuitous(0x0, {0xc000056fc0, 0x16})
        /src/pkg/vip/ndp.go:53 +0x110
github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).ensureIPAndSendGratuitous(0xc000418240, {0xc000184340, 0x6}, 0x0)
        /src/pkg/cluster/service.go:421 +0x3e5
github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).vipService.func3({0x2332fc0, 0xc000360140}, 0x0)
        /src/pkg/cluster/service.go:133 +0x26a
created by github.com/kube-vip/kube-vip/pkg/cluster.(*Cluster).vipService in goroutine 49
        /src/pkg/cluster/service.go:112 +0x9eb
```

**`Kube-vip.yaml`:**
```yaml
apiVersion: v1
kind: Pod
metadata:
  creationTimestamp: null
  name: kube-vip
  namespace: kube-system
spec:
  containers:
  - args:
    - manager
    env:
    - name: vip_arp
      value: "true"
    - name: port
      value: "6443"
    - name: vip_nodename
      valueFrom:
        fieldRef:
          fieldPath: spec.nodeName
    - name: vip_interface
      value: enp1s0
    - name: vip_cidr
      value: 32,128
    - name: dns_mode
      value: first
    - name: cp_enable
      value: "true"
    - name: cp_namespace
      value: kube-system
    - name: vip_leaderelection
      value: "true"
    - name: vip_leasename
      value: plndr-cp-lock
    - name: vip_leaseduration
      value: "15"
    - name: vip_renewdeadline
      value: "10"
    - name: vip_retryperiod
      value: "2"
    - name: address
      value: 10.10.10.12,fd00:4269:6943::12
    - name: prometheus_server
      value: :2112
    image: ghcr.io/kube-vip/kube-vip:v0.8.4
    imagePullPolicy: IfNotPresent
    name: kube-vip
    resources: {}
    securityContext:
      capabilities:
        add:
        - NET_ADMIN
        - NET_RAW
    volumeMounts:
    - mountPath: /etc/kubernetes/admin.conf
      name: kubeconfig
  hostAliases:
  - hostnames:
    - kubernetes
    ip: 127.0.0.1
  hostNetwork: true
  volumes:
  - hostPath:
      path: /etc/kubernetes/admin.conf
    name: kubeconfig
status: {}
```

**Additional context**
- Usage of super-admin.conf due to the kubernetes version and only on first control plane  (see issue #684).
- When reproducing the manifest with kube-vip v0.8.3 the pod runs flawlessly on every control plane.

BTW: thanks for the project and the great work :)

--- a/pkg/cluster/service.go
+++ b/pkg/cluster/service.go
@@ -108,7 +108,6 @@ func (cluster *Cluster) vipService(ctxArp, ctxDNS context.Context, c *kubevip.Co
 		}
 
 		if c.EnableARP {
-			// ctxArp, cancelArp = context.WithCancel(context.Background())
 			go func(ctx context.Context, i int) {
 				ipString := cluster.Network[i].IP()
 				isIPv6 := vip.IsIPv6(ipString)
@@ -130,7 +129,7 @@ func (cluster *Cluster) vipService(ctxArp, ctxDNS context.Context, c *kubevip.Co
 					case <-ctx.Done(): // if cancel() execute
 						return
 					default:
-						cluster.ensureIPAndSendGratuitous(cluster.Network[i].Interface(), ndp)
+						cluster.ensureIPAndSendGratuitous(i, ndp)
 					}
 					time.Sleep(3 * time.Second)
 				}
@@ -336,7 +335,7 @@ func (cluster *Cluster) StartLoadBalancerService(c *kubevip.Config, bgp *bgp.Ser
 						log.Debugf("(svcs) ending ARP update for %s via %s, every %dms", ipString, network.Interface(), c.ArpBroadcastRate)
 						return
 					default:
-						cluster.ensureIPAndSendGratuitous(network.Interface(), ndp)
+						cluster.ensureIPAndSendGratuitous(i, ndp)
 					}
 					if c.ArpBroadcastRate < 500 {
 						log.Errorf("arp broadcast rate is [%d], this shouldn't be lower that 300ms (defaulting to 3000)", c.ArpBroadcastRate)
@@ -390,45 +389,48 @@ func (cluster *Cluster) StartLoadBalancerService(c *kubevip.Config, bgp *bgp.Ser
 // ensureIPAndSendGratuitous - adds IP to the interface if missing, and send
 // either a gratuitous ARP or gratuitous NDP. Re-adds the interface if it is IPv6
 // and in a dadfailed state.
-func (cluster *Cluster) ensureIPAndSendGratuitous(iface string, ndp *vip.NdpResponder) {
-	for i := range cluster.Network {
-		ipString := cluster.Network[i].IP()
-		isIPv6 := vip.IsIPv6(ipString)
-		// Check if IP is dadfailed
-		if cluster.Network[i].IsDADFAIL() {
-			log.Warnf("IP address is in dadfailed state, removing [%s] from interface [%s]", ipString, iface)
-			err := cluster.Network[i].DeleteIP()
-			if err != nil {
-				log.Warnf("%v", err)
-			}
-		}
+func (cluster *Cluster) ensureIPAndSendGratuitous(index int, ndp *vip.NdpResponder) {
+	iface := cluster.Network[index].Interface()
+	ipString := cluster.Network[index].IP()
 
-		// Ensure the address exists on the interface before attempting to ARP
-		set, err := cluster.Network[i].IsSet()
+	// Check if IP is dadfailed
+	if cluster.Network[index].IsDADFAIL() {
+		log.Warnf("IP address is in dadfailed state, removing [%s] from interface [%s]", ipString, iface)
+		err := cluster.Network[index].DeleteIP()
 		if err != nil {
 			log.Warnf("%v", err)
 		}
-		if !set {
-			log.Warnf("Re-applying the VIP configuration [%s] to the interface [%s]", ipString, iface)
-			err = cluster.Network[i].AddIP(false)
-			if err != nil {
-				log.Warnf("%v", err)
-			}
-		}
+	}
 
-		if isIPv6 {
-			// Gratuitous NDP, will broadcast new MAC <-> IPv6 address
+	// Ensure the address exists on the interface before attempting to ARP
+	set, err := cluster.Network[index].IsSet()
+	if err != nil {
+		log.Warnf("%v", err)
+	}
+	if !set {
+		log.Warnf("Re-applying the VIP configuration [%s] to the interface [%s]", ipString, iface)
+		err = cluster.Network[index].AddIP(false)
+		if err != nil {
+			log.Warnf("%v", err)
+		}
+	}
+
+	if vip.IsIPv6(ipString) {
+		// Gratuitous NDP, will broadcast new MAC <-> IPv6 address
+		if ndp == nil {
+			log.Error("NDP responder was not created")
+		} else {
 			err := ndp.SendGratuitous(ipString)
 			if err != nil {
 				log.Warnf("%v", err)
 			}
-		} else {
-			// Gratuitous ARP, will broadcast to new MAC <-> IPv4 address
-			err := vip.ARPSendGratuitous(ipString, iface)
-			if err != nil {
-				log.Warnf("%v", err)
-			}
+		}
+
+	} else {
+		// Gratuitous ARP, will broadcast to new MAC <-> IPv4 address
+		err := vip.ARPSendGratuitous(ipString, iface)
+		if err != nil {
+			log.Warnf("%v", err)
 		}
 	}
-
 }

